aws/resource_aws_route53_resolver_firewall_domain_list.go

```diff
@@ -0,0 +1,189 @@
+	conn := meta.(*AWSClient).route53resolverconn
+
+	input := &route53resolver.CreateFirewallDomainListInput{
+		CreatorRequestId: aws.String(resource.PrefixedUniqueId("tf-r53-resolver-firewall-domain-list-")),
+		Name:             aws.String(d.Get("name").(string)),
+	}
+	if v, ok := d.GetOk("tags"); ok && len(v.(map[string]interface{})) > 0 {
+		input.Tags = keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().Route53resolverTags()
+	}
+
+	log.Printf("[DEBUG] Creating Route 53 Resolver DNS Firewall domain list: %#v", input)
+	output, err := conn.CreateFirewallDomainList(input)
+	if err != nil {
+		return fmt.Errorf("error creating Route 53 Resolver DNS Firewall domain list: %w", err)
+	}
+
+	d.SetId(aws.StringValue(output.FirewallDomainList.Id))
+	d.Set("arn", aws.StringValue(output.FirewallDomainList.Arn))
+
+	return resourceAwsRoute53ResolverFirewallDomainListUpdate(d, meta)
+}
+
+func resourceAwsRoute53ResolverFirewallDomainListRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).route53resolverconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	firewallDomainList, err := finder.FirewallDomainListByID(conn, d.Id())
+
+	if isAWSErr(err, route53resolver.ErrCodeResourceNotFoundException, "") {
+		log.Printf("[WARN] Route53 Resolver DNS Firewall domain list (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error getting Route 53 Resolver DNS Firewall domain list (%s): %w", d.Id(), err)
+	}
+
+	if firewallDomainList == nil {
+		log.Printf("[WARN] Route 53 Resolver DNS Firewall domain list (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	arn := aws.StringValue(firewallDomainList.Arn)
+	d.Set("arn", arn)
+	d.Set("id", firewallDomainList.Id)
+	d.Set("name", firewallDomainList.Name)
+
+	input := &route53resolver.ListFirewallDomainsInput{
+		FirewallDomainListId: aws.String(d.Id()),
+	}
+
+	domains := []*string{}
+
+	err = conn.ListFirewallDomainsPages(input, func(output *route53resolver.ListFirewallDomainsOutput, lastPage bool) bool {
+		domains = append(domains, output.Domains...)
+		return !lastPage
+	})
+
+	if err != nil {
+		return fmt.Errorf("error listing Route 53 Resolver DNS Firewall domain list (%s) domains: %w", d.Id(), err)
+	}
+
+	d.Set("domains", flattenStringSet(domains))
+
+	tags, err := keyvaluetags.Route53resolverListTags(conn, arn)
+	if err != nil {
+		return fmt.Errorf("error listing tags for Route53 Resolver DNS Firewall domain list (%s): %w", arn, err)
+	}
+
+	if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %w", err)
+	}
+
+	return nil
+}
+
+func resourceAwsRoute53ResolverFirewallDomainListUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).route53resolverconn
+
+	if v, ok := d.GetOk("domains"); ok && d.HasChange("domains") {
```

Comment:
Can all the domains for a list be deleted? If so will need to add a call to `UpdateFirewallDomains` with `Operation = REMOVE`.

Reply:
Updated - I'm now calling GetChanges to remove the old entries if there aren't any new entries anymore.